controllers/ssp_controller.go

```diff
@@ -153,8 +153,7 @@ func (r *sspReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ct
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
-
-	r.restartIfNeeded(instance)
+	restartNeeded := r.isRestartNeeded(instance)
```

Comment:
Can you move the declaration of `sspRequest` above this line, instead of splitting `r.restartIfNeeded()` into 2 functions?

Reply:
does this work? I want to check if the restart is needed before I clean the cache.

Reply:
Sure, you need to check the restart before cleaning cache.
What I meant is, you can move the definition of `sspRequest` variable before checking for restart:

```go
sspRequest := &common.Request{
	Request:        req,
	Client:         r.client,
	UncachedReader: r.uncachedReader,
	Context:        ctx,
	Instance:       instance,
	Logger:         reqLogger,
	VersionCache:   r.subresourceCache,
	TopologyMode:   r.topologyMode,
	CrdWatch:       r.crdWatch,
}

r.restartIfNeeded(instance)
r.clearCacheIfNeeded(instance)
```

Reply:
Nevermind, this would change logic.